lib/Fmt_ast.ml

```diff
@@ -2298,17 +2298,11 @@ and fmt_expression c ?(box = true) ?pro ?epi ?eol ?parens ?(indent_wrap = 0)
       in
       match compact with
       | None ->
-          (* TODO: remove leading_cmt once we figure out how to not introduce
-             regression with 4.07 Without the line below {[ let () = ( (*
-             before *) match (* after *) x with _ -> x) ]} Gets reformatted
-             into {[ let () = match (* before *) (* after *) x with _ -> x ]} *)
-          let leading_cmt = Cmts.fmt_before c e0.pexp_loc in
```

Comment:
This is still a concern. On 4.06.1, I get this diff on the test added in this PR:
```diff
-let _ =
-  (* before match *)
-  match (* after match *) x with _ -> 1
+let _ = match (* before match *)
+              (* after match *) x with _ -> 1
```

Reply:
I added a check on the compiler version, it's a bit too ad-hoc to my taste but we can think of a better design if the problem arises again somewhere else